vllm/model_executor/layers/vocab_parallel_embedding.py

```diff
@@ -60,31 +67,57 @@ def __init__(self,
         self.num_embeddings_padded = pad_vocab_size(num_embeddings,
                                                     padding_size)
         self.embedding_dim = embedding_dim
+
+        quant_method = quant_config.get_quant_method(
+            self) if quant_config else None
+        # lm_head may be quantized
+        self.linear_method = quant_method if quant_method is not None else UnquantizedLinearMethod(
+        )
+
         if params_dtype is None:
             params_dtype = torch.get_default_dtype()
+
         self.tp_size = get_tensor_model_parallel_world_size()
-        # Divide the weight matrix along the vocaburaly dimension.
+        # Divide the weight matrix along the vocabulary dimension.
         self.vocab_start_index, self.vocab_end_index = (
             vocab_range_from_global_vocab_size(
                 self.num_embeddings_padded, get_tensor_model_parallel_rank(),
                 self.tp_size))
         self.num_embeddings_per_partition = (self.vocab_end_index -
                                              self.vocab_start_index)
-        self.weight = Parameter(
-            torch.empty(self.num_embeddings_per_partition,
-                        self.embedding_dim,
-                        dtype=params_dtype))
-        set_weight_attrs(self.weight, {
-            "parallel_dim": 0,
-            "weight_loader": self.weight_loader
-        })
+
+        self.linear_method.create_weights(self,
+                                          self.embedding_dim,
+                                          [self.num_embeddings_per_partition],
+                                          self.embedding_dim,
+                                          self.num_embeddings_per_partition,
+                                          params_dtype,
+                                          weight_loader=self.weight_loader)
+
+        if not self.linear_method.QUANTIZED:
+            set_weight_attrs(self.weight, {
+                "parallel_dim": 0,
+            })
 
     def weight_loader(self, param: Parameter, loaded_weight: torch.Tensor):
-        parallel_dim = param.parallel_dim
-        assert loaded_weight.shape[parallel_dim] == self.org_vocab_size
-        loaded_weight = loaded_weight[self.vocab_start_index:self.
-                                      vocab_end_index]
-        param[:loaded_weight.shape[0]].data.copy_(loaded_weight)
+        if self.linear_method.QUANTIZED:
+            tp_rank = get_tensor_model_parallel_rank()
```

Comment:
Please use the existing pattern we have for weight loading of quantized weights with `packed_dim`

The pattern is that:
- In `create_weights`, we set the `packed_dim` dictionary as a `parameter` member
- In `weight_loader`, we look check if `packed_dim` is set. If so we adjust

Reply:
@robertgshaw2-neuralmagic  We copied the closest matching loader code from `MergedColumnParallelLinear` and copied that here. It worked for in our testing but likely not the best or the correct solution.